ivadomed/loader/loader.py

```diff
@@ -371,18 +375,28 @@ class MRI2DSegmentationDataset(Dataset):
         soft_gt (bool): If True, ground truths are expected to be non-binarized images encoded in float32 and will be
             fed as is to the network. Otherwise, ground truths are converted to uint8 and binarized to save memory
             space.
+        slice_filter_roi (bool): Indicates whether a slice filtering is done based on ROI data.
+        roi_thr (int): If the ROI mask contains less than this number of non-zero voxels, the slice will be discarded
+            from the dataset.
 
     """
 
     def __init__(self, filename_pairs, slice_axis=2, cache=True, transform=None, slice_filter_fn=None,
-                 task="segmentation", soft_gt=False):
+                 task="segmentation", roi_params=None, soft_gt=False):
         self.indexes = []
         self.filename_pairs = filename_pairs
         self.prepro_transforms, self.transform = transform
         self.cache = cache
         self.slice_axis = slice_axis
         self.slice_filter_fn = slice_filter_fn
         self.n_contrasts = len(self.filename_pairs[0][0])
+        if roi_params is None:
+            roi_params = {"suffix": None, "slice_filter_roi": None}
+        self.roi_thr = roi_params["slice_filter_roi"]
+        if roi_params["suffix"] is not None and isinstance(self.roi_thr, int):
```

Comment:
This could be simplified to `self.slice_filter_roi = roi_params["suffix"] is not None and isinstance(self.roi_thr, int)`

Reply:
zuper